From a76cbadd48237abdce0de9a1e3576b02e6111bf8 Mon Sep 17 00:00:00 2001
From: David Disseldorp <ddiss@suse.de>
Date: Wed, 20 Mar 2024 01:48:55 +0100
Subject: [PATCH] autofs-5.1.9 - fix ldap_parse_page_control() check

The final @cookie parameter should be a struct berval ** type. The
check currently fails when -Werror=incompatible-pointer-types is set:

conftest.c: In function 'main':
conftest.c:54:47: error: passing argument 4 of 'ldap_parse_page_control'
from incompatible pointer type [-Werror=incompatible-pointer-types]
   54 |       ret = ldap_parse_page_control(ld,clp,ct,c);
      |                                               ^
      |                                               |
      |                                               struct berval *
In file included from /usr/include/lber_types.h:24,
                 from /usr/include/lber.h:29,
                 from /usr/include/ldap.h:30,
                 from conftest.c:45:
/usr/include/ldap.h:2155:25: note: expected 'struct berval **' but
argument is of type 'struct berval *'
 2155 | ldap_parse_page_control LDAP_P((

Signed-off-by: David Disseldorp <ddiss@suse.de>
---
 aclocal.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/aclocal.m4 b/aclocal.m4
index 1046d72..fa18eb1 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -424,7 +424,7 @@ AC_LINK_IFELSE(
       #include <ldap.h> ]],
    [[ LDAP *ld;
       ber_int_t *ct;
-      struct berval *c;
+      struct berval **c;
       int ret;
       LDAPControl **clp;
       ret = ldap_parse_page_control(ld,clp,ct,c); ]])],
-- 
2.35.3

